lib/src/constants.dart

```diff
@@ -3,6 +3,25 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'package:analyzer/dart/constant/value.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+/// Throws an exception if [root] or its super(s) does not contain [name].
+void _assertHasField(ClassElement root, String name) {
+  var element = root;
+  while (element != null) {
+    final field = element.getField(name);
+    if (field != null) {
+      return;
+    }
+    element = element.supertype?.element;
+  }
+  final allFields = root.fields.toSet();
+  root.allSupertypes.forEach((t) => allFields.addAll(t.element.fields));
```

Comment:
[nit] https://www.dartlang.org/guides/language/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal

Could use a for loop, or I think it might still fit on 1 line to split this into 2 steps.

`root.allSupertype.map((t) => t.element.fields).forEach(allFields.addAll);`

or

`allFields.addAll(root.allSupertypes.expand((t) => t.element.fields));`